src/udp_tracker_connection.cpp

```diff
@@ -252,49 +252,11 @@ namespace libtorrent {
 			return;
 		}
 
-		m_target = pick_target_endpoint();
```

Comment:
`pick_target_endpoint()` seems to support the case `m_endpoints.empty()` as well. are you sure that can't happen?
there should probably be an assert for that.

Reply:
On the contrary, `pick_target_endpoint` assumes `!m_endpoints.empty()` by unconditionally dereferencing `m_endpoints.begin()` on its second line.

There is a check just above this line which returns if `m_endpoints.empty()` and a similar one above other other callsite so neither line can be reached if `m_endpoints.empty()`.